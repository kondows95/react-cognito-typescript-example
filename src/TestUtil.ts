import {ReactElement} from 'react';
import { render, fireEvent, RenderResult } from '@testing-library/react';
export default class TestUtil
{
  public render: RenderResult;
  constructor(component:ReactElement) {
    this.render = render(component);
  }

  setValue(testId: string, value: string | number) {
    const elem = this.render.getByTestId(testId);
    fireEvent.change(elem, {target: {value: value}})
  }

  click(testId: string) {
    const elem = this.render.getByTestId(testId);
    fireEvent.click(elem);
  }

  getAll(testId: string) {
    return this.render.getAllByTestId(testId);
  }

  get(testId: string) {
    return this.render.getByTestId(testId);
  }
}