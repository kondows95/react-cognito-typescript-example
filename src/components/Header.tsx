import React, {MouseEvent} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Grid, Box, Menu, MenuItem, AppBar, 
  IconButton, Toolbar, Typography, ListItemIcon, ListItemText
} from '@material-ui/core';
import { 
  Menu as MenuIcon,
  AccountBalanceWallet as LogoutIcon,
  MoreVert as MoreIcon,
} from '@material-ui/icons';
const ITEM_HEIGHT = 60;

const useStyles = makeStyles(theme => ({
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('xs')]: {
      display: 'block',
    },
  }
}));

type Props = {
  signOut: () => void;
}

const Header: React.FC<Props> = ({signOut}) => {
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null);
  const open = Boolean(anchorEl); 

  const handleLogout = () => {
    signOut();
  };

  const handleClose = () => {
    setAnchorEl(null);
  }

  const handleClick = (event: MouseEvent) => {
    setAnchorEl(event.currentTarget as HTMLButtonElement);
  }

  return (
    <AppBar position="fixed" className={classes.appBar}>
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="Open drawer"
          edge="start"
          //onClick={handleDrawerToggle}
          //disabled={(handleDrawerToggle === null)}
          className={classes.menuButton}
        >
          <MenuIcon />
        </IconButton>

        <Typography className={classes.title} variant="h6" noWrap>
          Management Console
        </Typography>
        
        <Box mr={0} ml="auto">
          <Grid container>
            <IconButton
              data-testid="openMenuButton"
              color="inherit"
              aria-label="More"
              aria-controls="long-menu"
              aria-haspopup="true"
              onClick={handleClick}
            >
              <MoreIcon />
            </IconButton>
            <Menu
              data-testid="long-menu"
              anchorEl={anchorEl}
              elevation={0}
              getContentAnchorEl={null}
              anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'center',
              }}
              transformOrigin={{
                vertical: 'top',
                horizontal: 'center',
              }}
              open={open}
              onClose={handleClose}
              PaperProps={{
                style: {
                  maxHeight: ITEM_HEIGHT * 4.5,
                  width: 200,
                },
              }}
            >
              <MenuItem 
                data-testid="signoutMenuItem" 
                onClick={handleLogout}
              >
                <ListItemIcon>
                  <LogoutIcon />
                </ListItemIcon>
                <ListItemText>Sing Out</ListItemText>
              </MenuItem>
              
            </Menu>
          </Grid>
        </Box>
      </Toolbar>
    </AppBar>
  );
};

export default Header;