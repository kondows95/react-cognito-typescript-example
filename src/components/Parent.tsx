import React, {PropsWithChildren} from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../store';

const Parent: React.FC<PropsWithChildren<{}>> = (props) => {
  return (
    <Provider store={store}>
        <Router>
          {props.children}
        </Router>
    </Provider>
  );
};
export default Parent;