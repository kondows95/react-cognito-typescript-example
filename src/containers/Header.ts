import { connect } from 'react-redux';
import Header from '../components/Header';
import { AppState, AppDispatch } from '../types';
import { signOut } from '../modules/auth';

export const mapStateToProps = (state: AppState) => ({
  authState: state.auth.authState,
  user: state.auth.user,
  loading: state.auth.loading,
});

export const mapDispatchToProps = (dispatch: AppDispatch) => ({
  signOut: () => dispatch(signOut())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
