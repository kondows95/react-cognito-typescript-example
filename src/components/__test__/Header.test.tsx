/*global jest,expect*/
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import Parent from '../Parent';
import Header from '../Header';
import TestUtil from '../../TestUtil';

let signOut : jest.Mock;
let util: TestUtil;
beforeEach(() => {
  signOut = jest.fn();
  util = new TestUtil(<Parent><Header signOut={signOut} /></Parent>);
});

it('Snapshot', () => {
  expect(util.render.asFragment()).toMatchSnapshot();
});

it('Open menu and click signout button', () => {
  util.click('openMenuButton');
  util.click('signoutMenuItem');
  expect(signOut).toHaveBeenCalled();   
});