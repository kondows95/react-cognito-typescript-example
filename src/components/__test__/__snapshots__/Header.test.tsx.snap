// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-1 MuiAppBar-colorPrimary mui-fixed"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="Open drawer"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h6 MuiTypography-noWrap"
      >
        Management Console
      </h6>
      <div
        class="MuiBox-root MuiBox-root-96"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <button
            aria-controls="long-menu"
            aria-haspopup="true"
            aria-label="More"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            data-testid="openMenuButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
